lib/spack/spack/bootstrap.py

```diff
@@ -354,9 +375,33 @@ def try_search_path(self, executables, abstract_spec_str):
 class _SourceBootstrapper(object):
     """Install the software needed during bootstrapping from sources."""
     def __init__(self, conf):
+        self.name = conf['name']
+        self.url = conf['info']['url']
+        self.metadata_dir = spack.util.path.canonicalize_path(conf['metadata'])
         self.conf = conf
         self.last_search = None
 
+    @property
+    def mirror_url(self):
+        # Absolute paths
+        if os.path.isabs(self.url):
+            return spack.util.url.format(self.url)
+
+        # Relative paths
+        if self.url.startswith('.'):
```

Comment:
Same thing here as for the BuildcacheBootstrapper.

Reply:
Done in d62c1a80